src/components/FilterPanel/FilterPanelCheckboxWithOverflowMenu/FilterPanelCheckboxWithOverflowMenu.js

```diff
@@ -0,0 +1,152 @@
+  const { createFocusHandler, createBlurHandler } = useComponentFocus(10);
+
+  const handleFocus = createFocusHandler(() => {
+    setShowOverflowButton(true);
+  });
+
+  const handleBlur = createBlurHandler(() => {
+    setShowOverflowButton(false);
+  });
+
+  /**
+   * Sets the width of the overflow menu to match the width of this component's width and adjusts
+   * its position so it is right alingned with the overflow menu button.
+   *
+   * @param {HTMLElement} menuBody Element containing the overflow menu options.
+   * @param {string} direction The menu position relative to the menu button. Not used.
+   * @param {HTMLElement} menuButton Overflow menu button element.
+   * @returns {{top: number, left: number}} The menu offset measurements.
+   */
+  const updateMenuWidthAndSetOffset = (menuBody, direction, menuButton) => {
+    const width = containerRef.current.clientWidth;
+    const left = width / 2 - width + menuButton.clientWidth / 2;
+
+    // Set the menu's width.
+    // eslint-disable-next-line no-param-reassign
+    menuBody.style.width = `${width}px`;
+
+    // Set the menu's left position to match the return left position value. We need to do this so
+    // that the overflow menu is initially positioned properly when the menu is too close to the
+    // edge of the viewport.
+    if (!overflowIsOpen) {
+      // eslint-disable-next-line no-param-reassign
+      menuBody.style.left = `${left}px`;
+    }
+
+    return {
+      top: 0,
+      left,
+    };
+  };
+
+  return (
+    <div
+      className={classnames(className, namespace, {
+        [`${namespace}--open`]: overflowIsOpen,
+      })}
+      onMouseOver={() => setShowOverflowButton(true)}
+      onMouseLeave={() => !overflowIsOpen && setShowOverflowButton(false)}
+      onFocus={handleFocus}
+      onBlur={handleBlur}
+      ref={containerRef}
+    >
+      <FilterPanelCheckbox
+        className={checkboxClassName}
+        wrapperClassName={classnames(
+          checkboxWrapperClassName,
+          `${namespace}__wrapper`
+        )}
+        {...other}
+      />
+      {showOverflowButton && (
+        <OverflowMenu
+          className={classnames(
+            overflowMenuClassName,
+            `${namespace}__overflow-button`
+          )}
+          menuOptionsClass={classnames(
+            overflowMenuOptionsClassName,
+            `${namespace}__overflow-options`
+          )}
+          menuOffsetFlip={updateMenuWidthAndSetOffset}
+          ariaLabel={overflowMenuAriaLabel}
+          onOpen={() => setOverflowIsOpen(true)}
+          onClose={() => setOverflowIsOpen(false)}
+          flipped
+        >
+          {children}
+        </OverflowMenu>
+      )}
+    </div>
+  );
+};
+
+FilterPanelCheckboxWithOverflowMenu.propTypes = {
+  ...FilterPanelCheckbox.propTypes,
+
+  /** Optional class name. */
+  className: PropTypes.string,
+
+  /** Optional checkbox class name. */
+  checkboxClassName: PropTypes.string,
```

Comment:
I'm curious about `checkboxClassName`, `checkboxWrapperClassName`, and `overflowMenuClassName` - these are definitely nice to have for additional customization, but do you foresee use cases in which this might be desired?

Reply:
That's a good question. I don't foresee that I'll be using them. Should I remove them?

Reply:
The biggest use case here is if someone wants to target a specific element, either for testing or their own customizations. Consider that they won't necessarily add an id to any of these inner elements -- they would need to reply on these class name props.

I vote that we keep them. IMO they definitely fill a need that we can't necessarily address otherwise.

Reply:
I think it's fair to remove them if we don't have an immediate need for it. It is easier to add customization later than remove it. I find that smaller component APIs are easier to read and make use of. For testing, consumers should not be relying on class names and there are ways to customize styles if necessary that don't require us to provide a prop for every single possible classname.